ssl/ssl_x509.cc

```diff
@@ -160,13 +160,15 @@ BSSL_NAMESPACE_BEGIN
 // installed. Calling an X509-based method on an |ssl| with a different method
 // will likely misbehave and possibly crash or leak memory.
 static void check_ssl_x509_method(const SSL *ssl) {
-  assert(ssl == NULL || ssl->ctx->x509_method == &ssl_crypto_x509_method);
+  assert(ssl == nullptr || (ssl != nullptr && ssl->ctx != nullptr &&
+                            ssl->ctx->x509_method == &ssl_crypto_x509_method));
 }
 
 // check_ssl_ctx_x509_method acts like |check_ssl_x509_method|, but for an
 // |SSL_CTX|.
 static void check_ssl_ctx_x509_method(const SSL_CTX *ctx) {
-  assert(ctx == NULL || ctx->x509_method == &ssl_crypto_x509_method);
+  assert(ctx == nullptr ||
+         (ctx != nullptr && ctx->x509_method == &ssl_crypto_x509_method));
```

Comment:
I don't think this change actually did anything -- the `ctx->x509_method == &ssl_crypto_x509_method` was already only being evaluated when `ctx != nullptr`.

Reply:
Ahh nice catch, I’ll just add the asserts in the designated functions we’re handling